JIB Image Build Regession in 1.4.0

Seems like there has been a bug/side effect introduced in 1.4.0 with build jib images.
 
The same image build that worked in versions before now gets an error when I run mvn install:

[ERROR] Failed to execute goal org.eclipse.jkube:kubernetes-maven-plugin:1.4.0:build (default) on project jkube-java-builder: Failed to execute the build: Error when building JIB image: Failed to create directory: /home/user/s2i-test/target/docker/test/1.0.0-SNAPSHOT/build/ -> [Help 1]

What is strange, is after that build, the folder has been created and if you run mvn install again it works.

The config is:

<images>
          <image>
                   <name>test:1.0.0-SNAPSHOT</name>

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -29,6 +29,7 @@ Usage:
 * Fix #904: `%g` should be resolved to namespace in OpenShift Maven Plugin
 * Fix #908: OpenShift-Maven-Plugin doesn't remove periods from container name
 * Fix #923: QuakusGenerator not applicable when using `io.quarkus.platform` groupId
+* Fix #895: FileUtil#createDirectory works for files with trailing separator (`/`)
 
 ### 1.4.0 (2021-07-27)
 * Fix #253: Refactor JKubeServiceHub's BuildService election mechanism via ServiceLoader

--- a/jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/FileUtil.java
+++ b/jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/FileUtil.java
@@ -247,11 +247,9 @@ public class FileUtil {
     }
 
     public static void createDirectory(File directory) throws IOException {
+      FileUtils.forceMkdir(directory);
         if (!directory.exists()) {
-            boolean isCreated = directory.mkdirs();
-            if (!isCreated) {
-                throw new IOException("Failed to create directory: " + directory.getAbsolutePath());
-            }
+          throw new IOException("Failed to create directory: " + directory.getAbsolutePath());
         }
     }
 }

--- a/jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/FileUtilTest.java
+++ b/jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/FileUtilTest.java
@@ -89,6 +89,15 @@ public class FileUtilTest {
     assertTrue(newDirectory.exists());
   }
 
+  // https://github.com/eclipse/jkube/issues/895
+  @Test
+  public void createDirectory_withTrailingSlash_shouldNotFail() throws IOException {
+    final File toCreate = new File(folder.getRoot().toPath().resolve("first").resolve("second").toFile(),
+        File.separator);
+    FileUtil.createDirectory(toCreate);
+    assertTrue(toCreate.exists());
+  }
+
   @Test
   public void testListFilesRecursively() throws IOException {
     prepareDirectory();

